LINK-1646 | Bump HDS to version 3.1.0 and update header

jorilindell opened the pull request:
## Description
- Bump HDS to version 3.1.0
- Change HDS Navigation component to Header component

## Closes
[LINK-1646](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1646)

## Screenshot
<img width="1437" alt="Screenshot 2023-10-17 at 14 04 37" src="https://github.com/City-of-Helsinki/linkedregistrations-ui/assets/24706814/cef09002-5fac-42c9-bc10-b36ed78a4b47">


[LINK-1646]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1646?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

arpargo commented:
Looks good, seems to mostly work.
Two things I noticed:
a) The close text in right corner of the screenshot is clearly missing translations (was there on testing aswell)
b) For some reason, when on the base url (localhost:3001) the finnish language doesn't work locally? Not sure if this is a feature or not but thought I should mention it. Translations work properly on other 404 pages like '/asd'.

jorilindell commented:
> Looks good, seems to mostly work. Two things I noticed: a) The close text in right corner of the screenshot is clearly missing translations (was there on testing aswell) b) For some reason, when on the base url (localhost:3001) the finnish language doesn't work locally? Not sure if this is a feature or not but thought I should mention it. Translations work properly on other 404 pages like '/asd'.

- Added translations for the close text
- Added prefix to default locale to avoid problems with automatic language detection. See https://nextjs.org/docs/pages/building-your-application/routing/internationalization#prefixing-the-default-locale

harriris-vincit commented on public/locales/sv/common.json:
```diff
@@ -1,8 +1,10 @@
 {
   "appName": "Linked Registrations",
+  "ariaLogo": "Service logo",
```
I think "Tjänstens logo" is the proper Swedish translation

jorilindell replied:
Changed

harriris-vincit reviewed (approved):
Looks good
